### PR TITLE
feat(browser): sign out

### PR DESCRIPTION
**Pull request by demonzoo**
## Summary

Provide sign out method in browser sdk

## Linear Issue Reference

LOG-793

## Testing

Test cases will be provided in another PR that links to LOG-1245

### CODE REVIEW

File: packages/browser/src/index.ts [hunk 1]
```diff
@@ -68,6 +70,34 @@ export default class LogtoClient {
     window.location.assign(signInUri);
   }
 
+  public async signOut(postLogoutRedirectUri?: string) {
+    if (!this.idToken || !this.oidcConfig) {
+      return;
```
Comment (wangsijie) [inline packages/browser/src/index.ts:75]: maybe throw an error?

Comment (demonzoo) [reply packages/browser/src/index.ts:75]: browser package doesn't have error handling utilities at this point, will throw error once #175 is merged.

**Review by wangsijie — APPROVED** [review]

File: packages/browser/src/index.ts [hunk 2]
```diff
@@ -68,6 +70,34 @@ export default class LogtoClient {
...
+      void revoke(revocationEndpoint, clientId, this.refreshToken, requester);
+    }
+
+    const url = generateSignOutUri({
+      endSessionEndpoint,
+      postLogoutRedirectUri,
+      idToken: this.idToken,
+    });
+
+    window.location.assign(url);
```
Comment (simeng-li) [inline packages/browser/src/index.ts:92]: nit: `window.location.assign(url);` could leave it as the last statement.

**Review by simeng-li — APPROVED** [review]
LGTM

File: packages/browser/src/index.ts (hunk 2 above)
Comment (demonzoo) [reply packages/browser/src/index.ts]: Code updated.

File: packages/browser/src/index.ts (hunk 1 above)
Comment (gao-sun) [reply packages/browser/src/index.ts:75]: should to await oidc config instead of directly returning.

File: packages/browser/src/index.ts [hunk 3]
```diff
@@ -68,6 +70,35 @@ export default class LogtoClient {
...
+    if (!this.idToken || !this.oidcConfig) {
+      return;
+    }
+
+    const { clientId, requester } = this.logtoConfig;
+    const { endSessionEndpoint, revocationEndpoint } = this.oidcConfig;
+    const logtoKey = getLogtoKey(clientId);
+
+    if (this.refreshToken) {
+      void revoke(revocationEndpoint, clientId, this.refreshToken, requester);
```
Comment (gao-sun) [inline packages/browser/src/index.ts:83]: this is not good since the request will be interrupted when `window.location` changed

**Pull request merged**